app/src/behaviors/behavior_backlight.c

```diff
@@ -20,21 +20,59 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
 static int behavior_backlight_init(const struct device *dev) { return 0; }
 
+static int
+on_keymap_binding_convert_central_state_dependent_params(struct zmk_behavior_binding *binding,
+                                                         struct zmk_behavior_binding_event event) {
+    switch (binding->param1) {
+    case BL_TOG_CMD: {
+        bool state;
+        int err = zmk_backlight_get_state(&state);
+        if (err) {
+            LOG_ERR("Failed to get backlight state (err %d)", err);
```

Comment:
The function `zmk_backlight_get_state` never returns an error so this check may be removed. Maybe is also worth to use the old singature:  `bool zmk_backlight_is_on();`.

Reply:
Ok will sort, not aure about `backlight_is_on` as opposed to get state but will change if you feel strongly about it

Reply:
If the function never returns an error, it's preferred to directly return the value, for example:
https://github.com/zmkfirmware/zmk/blob/main/app/include/zmk/usb.h#L24
https://github.com/zmkfirmware/zmk/blob/main/app/include/zmk/wpm.h#L9
https://github.com/zmkfirmware/zmk/blob/main/app/include/zmk/battery.h#L9

Reply:
As for the name, I think `get_state` is a bit ambiguous since in the code `state` refers to the whole state (brightness and on/off).

Reply:
The name `get_on` is ok, but I still think it should return the value directly in the same way as `get_brt`